### DISCUSSION TITLE
Way to maintain api compatibility between different projects/types of project

### DISCUSSION THREAD

**Discussion by Artur2** [dotnet/runtime#53071] [Q&A]
Hi, everybody!
I have a question about versioning and how you do that with so many different kinds of projects and dependencies
For example: 
Microsoft maintaining WPF library project, which depends on .net runtime
So how you follow api compatibility when working on feature for WPF library ?
Just reading breaking changes ?

I know you use semver in some projects, for example in Visual Studio Code
Where you releasing minor version with requested/planned features, than sometimes(if you have bugs/issues) working on "Endgame" where you resolving issues, bugs

As I know, in Visual Studio Code now only 1.*.* major version, it's look like there is no breaking changes since first release
But for other products, like Entity Framework Core you change API, what sometimes following to breaking changes, or some package which uses EF Core introducing breaking changes, so how you resolve that ?

Or may be you have some cheat sheet or practices to avoid frequent issues in API changes, which breaks for example front-end application and back-end application compatibility (front: angular/react/ect, backend: ASP.NET Core/etc)

I'm not sure about right place for this question, anyway open for suggestions from you

**Comment by huoyaoyuan** [comment]
Generally, personal project seldom grows enough to worry about versioning. I often use 0.x.y.

**Comment by Artur2** [comment]
Hm... Seems to be wrong repository for that discussion 😿
